### PR TITLE
Eliom_client.History: give access to visited URLs

### PR DESCRIPTION
**Pull request by jrochel**

### CODE REVIEW

File: src/lib/eliom_client.client.ml [hunk 1]
```diff
@@ -536,6 +536,29 @@ module HistCache = Map.Make (struct
   let compare = compare
 end)
 
+module History = struct
+  let urls = ref HistCache.empty
+
+  let push_current_url () =
+    let page = !active_page in
+    let id = page.page_id.state_index in
+    let url = Js.to_string Dom_html.window##.location##.href in
```
Comment (vouillon) [inline src/lib/eliom_client.client.ml:545]: The location is not changed in the app:
https://github.com/ocsigen/eliom/blob/50e207e76cc09d0f0491be6b4eab9c14f600a77f/src/lib/eliom_client.client.ml#L671-L672

**Comment by jrochel** [comment]
@vouillon Tested.

File: src/lib/eliom_client_core.client.ml [hunk 2]
```diff
@@ -715,26 +715,34 @@ end
...
-  set_page_status p @@ if !(p.page_is_cached) then Cached else Dead
+  set_page_status p @@ match p.dom with Some _ -> Cached | None -> Dead
 
 let session_id = random_int ()
 let next_state_id =
   let last = ref 0 in fun () -> incr last; {session_id; state_index = !last}
 
 let last_page_id = ref (-1)
-let mk_page ?(state_id = next_state_id ()) ~status () =
+let mk_page ?(state_id = next_state_id ()) ?url ?previous_page ~status () =
```
Comment (vouillon) [inline src/lib/eliom_client_core.client.ml:745]: It seems the `url` argument is not used

File: src/lib/eliom_client_core.client.ml [hunk 3]
```diff
@@ -666,7 +666,7 @@ let rec rebuild_rattrib node ra = match Xml.racontent ra with
 
 
 
-(* == Associate data to state of the History API.
+(* == Associate data to state of the istory API.
```
Comment (Zzull) [inline src/lib/eliom_client_core.client.ml:669]: Probably better with an `h`

**Pull request merged**